### ISSUE TITLE
请教下关于MaskedGenerativeEncoderViT网络设计的问题

### ISSUE THREAD

**Issue by Harxis** [HaiyuWu/Vec2Face#17]
请问，在 `MaskedGenerativeEncoderViT` 的 `forward_encoder` 函数里，模型代码 `model_vec2face.py` 的第 238 行：

```python
feature_token = torch.cat([encode_feature.unsqueeze(1), feature_token], dim=1)
```

为什么要把 `encode_feature` 拼接到 `feature_token` 的最前面，作为第 0 个 token？

* 我看到在 masking 操作中，这个第 0 号 token 并不会被丢弃或遮盖。
* `encode_feature` 本身已经包含了完整的人脸表征向量，是输入的全局先验。

这样设计是否会让模型直接利用全局向量来生成或重建人脸，而不真正学习各个局部 token 的特征？
请问这样做的主要动机是什么？有没有可能因此影响到 Masked Autoencoder 原本对局部信息的学习？

**Comment by HaiyuWu** [comment]
你好，

我没有研究它具体的影响，不过这个操作会加快模型收敛速度。主要的动机是为了在后续fmae中可以用这个feature来填masked positions来保证ID的准确性。
不过这个操作很有可能会影响局部信息的学习，因为就生成出来的图像来，确实缺少一些人脸属性的多样性。这是一个值得深入探讨的点。

**Comment by Harxis** [comment]
好的，感谢你的回复。
